lib/ddtrace/configuration/settings.rb

```diff
@@ -42,19 +44,44 @@ def initialize(*_)
         tags
       end
 
+      # Legacy [App Analytics](https://docs.datadoghq.com/tracing/legacy_app_analytics/) configuration.
+      #
+      # @deprecated Use [Trace Retention and Ingestion](https://docs.datadoghq.com/tracing/trace_retention_and_ingestion/)
+      #   controls.
+      # @public_api
       settings :analytics do
+        # @default `DD_TRACE_ANALYTICS_ENABLED` environment variable, otherwise `nil`
+        # @return [Boolean,nil]
         option :enabled do |o|
           o.default { env_to_bool(Ext::Analytics::ENV_TRACE_ANALYTICS_ENABLED, nil) }
           o.lazy
         end
       end
 
+      # Profiler API key.
+      # @default `DD_API_KEY` environment variable, otherwise `nil`
+      # @return [String,nil]
+      # @public_api
       option :api_key do |o|
         o.default { ENV.fetch(Ext::Environment::ENV_API_KEY, nil) }
         o.lazy
       end
 
+      # Internal tracer diagnostic settings.
+      #
+      # Enabling these surfaces debug information that can be helpful to
+      # diagnose issues related to the tracer internals.
+      # @public_api
       settings :diagnostics do
+        # Outputs all spans created by the host application to `Datadog.logger`.
+        #
+        # **This option is very verbose!** It's only recommended for non-production
+        # environments.
+        #
+        # This option is helpful when trying to understand what information the
+        # tracer is sending to the Agent or backend.
+        # @default `DD_TRACE_DEBUG` environment variable, otherwise `false`
+        # @return [Boolean]
         option :debug do |o|
```

Comment:
This actually enables debug logging on the default logger, so it applies to all products, not just tracer.

(I wish we had a way to enable some debug logging without the full trace output, since it gets really noisy while debugging >_>)

Reply:
>(I wish we had a way to enable some debug logging without the full trace output, since it gets really noisy while debugging >_>)

Debug levels? :)

Or we can add a `span_debug` (wip name) option that controls only span output. I agree that span output is much more verbose than anything else.

I guess it would be similar to the profiler outputting all its collected stacks.

Reply:
Log4j jokes aside, I've spent more time even in the Java ecosystem than in the Ruby ecosystem, and I think Java logging frameworks solve it in quite a reasonable way -- by creating a new logger instance per class:

```java

Logger logger = Logger.getLogger(MyClass.class.getName());
```

Then, most logging frameworks allow you to configure per-class (or even per namespace) logging levels, so solving this "issue" in would be as simple as enabling debug logging for everything other than the class that prints out the full spans.

The default Ruby logger does not have this behavior, but there are some third party logger gems that do, for instance [logging](https://github.com/twp/logging) and [semantic logger](https://logger.rocketjob.io/).

So I'd actually lean into either using a better 3rd-party logger or enhancing the one we already use to get these kinds of behaviors, rather than having ad-hoc options for some of the more noisy things.